### PR TITLE
Containerize rapid tools

### PR DESCRIPTION
**Pull request by badosu**
To create an image, install docker-compose, clone the repository and run `docker-compose up`

To refresh the served assets run: `docker-compose exec rapid <rapid-update-git.sh|rapid-update-repos.py>`

### CODE REVIEW

**Comment by badosu** [comment]
cc @lhog

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+version: "3.9"
+services:
+  rapid:
+    build:
+      context: .
+      target: scratch
+      args:
+        RAPID_GIT: /rapid-git
+        RAPID_PACKAGES: /rapid-packages
+        RAPID_GIT_REPOS: https://github.com/springraaar/metal_factions
```
Comment (lhog) [inline docker-compose.yml:10]: Is it a leftover/example or is it intended?

File: src/BuildGit.cpp [hunk 2]
```diff
@@ -278,7 +278,8 @@ void buildGit(
 	std::string const & Prefix)
 {
 	// Initialize libgit2
-	checkRet(git_libgit2_init(), "git_libgit2_init()");
+	//checkRet(git_libgit2_init(), "git_libgit2_init()");
```
Comment (lhog) [inline src/BuildGit.cpp:281]: What is wrong with `checkRet()` ?

Comment (badosu) [reply src/BuildGit.cpp:281]: The input is returning 1 for some reason, I didn't investigate why yet

File: docker-compose.yml (hunk 1 above)
Comment (badosu) [reply docker-compose.yml:10]: This is for testing only, I don't intend to merge this PR until it's on a deployable state. mf is smaller than byar for testing

**Comment by lhog** [comment]
Let's do rebase first so changes are easier to distill:

* Please make another branch, a clone of this one, reset it back to the https://github.com/beyond-all-reason/pr-downloader/pull/1/commits/b1fcc68a7c053274143ed81c3c47c6862b9fc67e and make another PR
* Once merged rebase this one to the (future) HEAD of BAR/pr-d and push-force

**Comment by lhog** [comment]
Thanks, feel free to rebase.

File: src/BuildGit.cpp (hunk 2 above)
Comment (Beherith) [reply src/BuildGit.cpp:281]: Yeah its been annoying as hell

**Pull request closed without merging**